### PR TITLE
docs: Release notes for 0.13.0.

### PR DESCRIPTION
**Pull request by determined-dsw**
## Commentary (optional)

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: docs/release-notes.txt [hunk 1]
```diff
@@ -3,6 +3,134 @@
...
+------------
+
+Version 0.13.0
+^^^^^^^^^^^^^^
+
+**Release Date:** August 20, 2020
+
+**New Features**
+
+- Allow experiments to connect to the master using TLS.
```
Comment (neilconway) [inline docs/release-notes.txt:16]: "Experiments" aren't really the things connecting to the master -- "trials"? "Training workloads"?

File: docs/release-notes.txt [hunk 2]
```diff
@@ -3,6 +3,134 @@
...
+
+- Allow agent's TLS verification to skip verification or use a custom certificate for the master.
+
+- For ``TFKerasTrial`` and ``EstimatorTrial``, add functionality to ``context.wrap_dataset()`` which allows users to skip sharding their training dataset for distributed training. If using this option, users are responsible for splitting their dataset in such a manner that every GPU (rank) sees unique data.
```
Comment (neilconway) [inline docs/release-notes.txt:20]: A bit more detail on how you might use this "functionality" would be nice -- e.g., is it activated by passing a particular argument to wrap_dataset?

Also, let's link to the API docs for wrap_dataset.

File: docs/release-notes.txt [hunk 3]
```diff
@@ -3,6 +3,134 @@
...
 
+Version 0.13
+------------
+
+Version 0.13.0
+^^^^^^^^^^^^^^
+
+**Release Date:** August 20, 2020
+
+**New Features**
```
Comment (neilconway) [inline docs/release-notes.txt:14]: Should we include up-top a note that all active experiments must be paused and/or terminated (not sure which) before starting an upgrade?

File: docs/release-notes.txt [hunk 4]
```diff
@@ -3,6 +3,134 @@
...
+
+- For ``TFKerasTrial`` and ``EstimatorTrial``, add functionality to ``context.wrap_dataset()`` which allows users to skip sharding their training dataset for distributed training. If using this option, users are responsible for splitting their dataset in such a manner that every GPU (rank) sees unique data.
+
+**Improvements**
+
+- **Breaking Change:** Change how additional arguments to ``ssh`` are passed through ``det shell start`` and ``det shell open``.  Previously they were passed as a single string, like ``det shell open SHELL_ID --ssh-opt '-X -Y -o SomeSetting="some string"'``, but now the ``--ssh-opt`` has been removed and all extra positional arguments are passed through without requiring double-layers of quoting, like ``det shell open SHELL_ID -- -X -Y -o SomeSetting="some string"`` (note the use of ``--`` to indicate all following arguments are positional arguments).
+
+- Install the ``determined`` and ``determined_cli`` Python packages inside ``det shell``.  Also, pass any user-defined environment variables for the task image into the ssh sessions opened via ``det shell start`` or ``det shell open``.
```
Comment (neilconway) [inline docs/release-notes.txt:26]: `determined-cli`, I think.

File: docs/release-notes.txt [hunk 5]
```diff
@@ -3,6 +3,134 @@
...
+
+- **Remove Determined Steps from the UX:** Remove the concept of a "Determined step" from the UX and add new configuration settings to allow settings previously in terms of steps to be configured instead in terms of records, batches or epochs. See the :ref:`migration guide <migration-guide_remove-steps>` for details on migrating from the old configuration to the new configuration.
```
Comment (neilconway) [inline docs/release-notes.txt:28]: s/Determined step/step/g

Comment (vishnu2kmohan) [inline docs/release-notes.txt:28]: suggestion: @stoksc we should draw attention to the fact that this is a breaking and disruptive change.

File: docs/release-notes.txt [hunk 6]
```diff
@@ -3,6 +3,134 @@
...
+
+  - Many configuration settings can now be set in terms of records, batches or epochs. For example, a single searcher can be configured to run for 100 records by setting ``max_length: {records: 100}``, 100 batches by setting ``max_length: {batches: 100}``, or 100 epochs by setting ``records_per_epoch`` at the root of the config and ``max_length: {epochs: 100}``.
+  - A new configuration setting, ``records_per_epoch``, is added that must be specified when any quantity is configured in terms of epochs.
```
Comment (neilconway) [inline docs/release-notes.txt:31]: Here and below, let's make the tense consistent -- we don't normally write in past tenses for changes in the current release ("was replaced" -> "has been replaced" or "is replaced by")

File: docs/release-notes.txt [hunk 7]
```diff
@@ -3,6 +3,134 @@
...
+    - Limit logs to 1000 lines for initial load and load additional 1000 for each subsequent older log fetches.
+    - Use new log viewer optimized for efficient rendering.
+    - Introduce log line numbers.
+    - Add ANSI color support.
+    - Add error, warning, and debug visual icons and colors.
+    - Add tailing button to enable tailing lob behavior.
+    - Add scroll to top button to load older logs out
+    - Fix back and forth scrolling behavior on log viewer.
+
+  - Cluster: ``/det/Cluster``
```
Comment (neilconway) [inline docs/release-notes.txt:88]: I'd assume it is `/det/cluster`?

File: docs/release-notes.txt [hunk 8]
```diff
@@ -3,6 +3,134 @@
...
+
+- Security: Apply user authentication to streaming endpoints.
+
+- Shell: Support ``det shell`` in host networking mode.
+
+- Shell: Support ``det shell`` in cloud environments.
+
+- Tasks: Allow the master certificate file to be readable even for a non-root task.
+
+- TensorBoard: Fix issue affecting TensorBoards on AWS in US-EAST-1 region.
```
Comment (vishnu2kmohan) [inline docs/release-notes.txt:120]: non-blocking: Lowercase `us-east-1`

File: docs/release-notes.txt (hunk 5 above)
Comment (stoksc) [reply docs/release-notes.txt:28]: Yeah sure, let's drop **Breaking Change:** in front of it?

File: docs/release-notes.txt [hunk 9]
```diff
@@ -3,6 +3,134 @@
...
+
+    - State filter converted from multi-select to single-select.
+    - Convert actions from expanded buttons to overflow menu (triple vertical dots).
+    - Batch operation logic change to available if the action can be applied to any of the selected experiments
+    - Add pagination support that auto turns on when entries extend beyond 10 entries.
+
+  - Experiment Detail: ``/det/experiments/<id>``
+
+    - Implement charting with plotly with zooming capability.
+    - Trial table paginates on the webui side in preparation for API pagination in the near future.
```
Comment (hamidzr) [inline docs/release-notes.txt:60]: nit: WebUI

File: docs/release-notes.txt [hunk 10]
```diff
@@ -3,6 +3,134 @@
...
+    - Convert steps to batches in trials table and metric chart.
+    - Update continue trial flow to use batches, epochs or records.
+    - Use Monaco editor for the experiment config with YAML syntax highlighting.
+    - Add links to source for Checkpoint modal view, allowing users to navigate to the corresponding experiment or trial for the checkpoint.
+
+  - Trial Detail: ``/det/trials/<id>``
+
+    - Add trial information table.
+    - Add trial metrics chart.
+    - Implement charting with plotly with zooming capability.
```
Comment (hamidzr) [inline docs/release-notes.txt:70]: nit: Plotly

File: docs/release-notes.txt [hunk 11]
```diff
@@ -3,6 +3,134 @@
...
+  - Experiment List: ``/det/experiments``
+
+    - State filter converted from multi-select to single-select.
+    - Convert actions from expanded buttons to overflow menu (triple vertical dots).
+    - Batch operation logic change to available if the action can be applied to any of the selected experiments
+    - Add pagination support that auto turns on when entries extend beyond 10 entries.
+
+  - Experiment Detail: ``/det/experiments/<id>``
+
+    - Implement charting with plotly with zooming capability.
```
Comment (hamidzr) [inline docs/release-notes.txt:59]: nit: Plotly

File: docs/release-notes.txt (hunk 3 above)
Comment (stoksc) [reply docs/release-notes.txt:14]: They must be terminated. There's a warning at the top of the migration guide, but two warnings couldn't hurt.

File: docs/release-notes.txt [hunk 12]
```diff
@@ -3,6 +3,134 @@
...
+  - For single, random and grid searchers ``searcher.max_steps`` was replaced by ``searcher.max_length``
+  - For ASHA based searchers, ``searcher.target_trial_steps`` and ``searcher.step_budget`` were replaced by ``searcher.max_length`` and ``searcher.budget``, respectively.
+  - For PBT, ``searcher.steps_per_round`` was replaced by ``searcher.length_per_round``.
+  - For all experiments, the names for ``min_validation_period`` and ``min_checkpoint_period`` are unchanged but they are now configured in terms of records, batches or epochs.
+
+- **WebUI changes**
+
+  - Tasks List: ``/det/tasks``
+
+    - Consolidate notebooks, tensorboards, shells, commands into single list page.
```
Comment (hamidzr) [inline docs/release-notes.txt:41]: nit: capitalize notebook and tensorboard

File: docs/release-notes.txt [hunk 13]
```diff
@@ -3,6 +3,134 @@
...
+    - Implement charting with plotly with zooming capability.
+    - Trial info table paginates on the webui side in preparation for API pagination in the near future.
+    - Add support for batches, records and epochs for experiment config.
+    - Convert metric chart to show batches.
+    - Convert steps table to batches table.
+
+  - Master Logs: ``/det/logs``, Trial Logs: ``/det/trials/<id>/logs``, Task Logs: ``/det/<tasktype>/<id>/logs``
+
+    - Logs now stream in new logs events on the fly.
+    - Limit logs to 1000 lines for initial load and load additional 1000 for each subsequent older log fetches.
```
Comment (hamidzr) [inline docs/release-notes.txt:79]: nit: an additional

File: docs/release-notes.txt [hunk 14]
```diff
@@ -3,6 +3,134 @@
...
+    - Add trial metrics chart.
+    - Implement charting with plotly with zooming capability.
+    - Trial info table paginates on the webui side in preparation for API pagination in the near future.
+    - Add support for batches, records and epochs for experiment config.
+    - Convert metric chart to show batches.
+    - Convert steps table to batches table.
+
+  - Master Logs: ``/det/logs``, Trial Logs: ``/det/trials/<id>/logs``, Task Logs: ``/det/<tasktype>/<id>/logs``
+
+    - Logs now stream in new logs events on the fly.
```
Comment (hamidzr) [inline docs/release-notes.txt:78]: we don't have streaming endpoints for task types yet.

File: docs/release-notes.txt [hunk 15]
```diff
@@ -3,6 +3,134 @@
...
+    - Limit logs to 1000 lines for initial load and load additional 1000 for each subsequent older log fetches.
+    - Use new log viewer optimized for efficient rendering.
+    - Introduce log line numbers.
+    - Add ANSI color support.
+    - Add error, warning, and debug visual icons and colors.
+    - Add tailing button to enable tailing lob behavior.
```
Comment (hamidzr) [inline docs/release-notes.txt:84]: I don't think these are new to this release cc @hkang1

Comment (hkang1) [reply docs/release-notes.txt:84]: @hamidzr is right. The logs changes were previously applied to Master logs and trial logs, but now they have been applied to task logs. I don't think it hurts have them here again due to the nature of the release.

Comment (hkang1) [inline docs/release-notes.txt:84]: **nit:** oops, should say `log` and not `lob`

**Pull request merged**